model/DataStore/ServiceProvider/DataStoreServiceProvider.php

```diff
@@ -36,6 +36,9 @@ public function __invoke(ContainerConfigurator $configurator): void
     {
         $services = $configurator->services();
 
+        /**
+         * @deprecated use oat\tao\model\metadata\compiler\AdvancedJsonResourceMetadataCompiler
+         */
```

Comment:
Instead of this, you can call `->deprecate()`:
```php
$services
    ->set(JsonMetadataCompiler::class, JsonMetadataCompiler::class)
    ->deprecate(/* string $package, string $version, string $message */)
```

where
`$package` - the name of the composer package that is triggering the deprecation
`$version` - the version of the package that introduced the deprecation
`$message` - the deprecation message to use